src/createAnimatedComponent/createAnimatedComponent.tsx

```diff
@@ -87,15 +87,7 @@ type Options<P> = {
  * @see https://docs.swmansion.com/react-native-reanimated/docs/core/createAnimatedComponent
  */
 
-/**
- * @deprecated Please use `Animated.FlatList` component instead of calling `Animated.createAnimatedComponent(FlatList)` manually.
- */
-// @ts-ignore This is required to create this overload, since type of createAnimatedComponent is incorrect and doesn't include typeof FlatList
-export function createAnimatedComponent(
-  component: typeof FlatList<unknown>,
-  options?: Options<any>
-): ComponentClass<AnimateProps<FlatListProps<unknown>>>;
-
+// Don't change the order of overloads, since such a change breaks current behavior
```

Comment:
It appears to be a flawed approach when our only means of ensuring that we don't break anything is through a comment. Is it possible to implement some TypeScript tests instead? I believe that @tjzel has already worked on something similar.

Reply:
Unfortunately there are no tests for deprecation